### DISCUSSION TITLE
Including Service records in the Datahub

### DISCUSSION THREAD

**Discussion by jahow** [geonetwork/geonetwork-ui#565] [Ideas]
Currently the Datahub application only shows records which are classified as datasets.

Several people have asked whether showing service records could make sense as well. I'm not convinced that showing services alongside datasets in the results would make a lot of sense as both kinds have quite different uses and audiences.

One possibility could be to add a "Services" tab between Datasets and Organizations. Then the Service tab would offer a search interface with slightly different options, as well as a different full view page. The Service view page could for example dynamically show a list of available layers, output formats etc.

Showing links between services and datasets when they exist could also be interesting.

Looking for opinions, thanks!!

**Comment by lucilecollignon** [comment]
If it turns out that most users come looking for either a dataset or a service, then there are 2 possibilities:
- Display them in different tabs: Dataset tab and Services tab.
- The other possibility is to add a selector inside the search field to search either only datasets, only services, or both.

In all cases, you will need to create a specific layout for the template page for a service, and distinguish between the 2 types of thumbnail.

In any case, showing links between services and datasets when they exist seems a good idea to me.

**Reply by IGNF-Xavier** [reply]
I don't know which possibilities the users want.
The first one seems to be easier to implement.
In the two possibilities, it will need a configurable template to show correctly the stored metadata (not only dataset and services)

**Comment by fvanderbiest** [comment]
Thanks Olivia for bringing the subject to light.
I'm really not sure that end users expect services to be displayed. Most of them will not know what to do with it. So, it rather looks like a niche to me, and I would treat it as such: for any search that brings services with the datasets, I would display a small string like "this search also matches several data services - click to display".

**Reply by fvanderbiest** [reply]
I also would like to say that it brings an additional layer of complexity to the app, with little added value IMO.

**Reply by IGNF-Xavier** [reply]
Today services of many platform are limited to WFS, TMS, WMTS, WMS...
But there is also geocoding, itinerary, search, etc and it will be useful to have metadata of these services.

**Reply by jahow** [reply]
Right, these services could be useful to users! but in that case it might make sense to show them in a separate tab or even a separate app, and present them in a way that's less metadata-oriented and more usage-oriented: give instructions, offer sample requests and examples, and links to reuses.

**Comment by jahow** [comment]
Another option would be to show the services as part of the authenticated backoffice of the datahub; then we would be able to show a slightly more technical view of the metadata content, e.g. available layers and protocols in the service.

Maybe this would make more sense in terms of the intended audience @fvanderbiest?

**Reply by fvanderbiest** [reply]
Indeed.

**Comment by fgravin** [comment]
For the record [Geocat.ch](https://www.geocat.ch/datahub/search?q=ogc%20wms) asked to add the services as well.
As it's a fork, I just removed the filter. It "could" be customisable but indeed it changes a bit the scope of the datahub, so should be analyzed in the UX way :smile: 

Note that it is an INSPIRE requirement though.
